Feature/block1 Setup Azure DevOps project (#18)
devonfw/hangar: documentation/azure-devops/setup-project.asciidoc

== *Creating the Azure DevOps project*

There are two ways of creating an Azure DevOps project:
[loweralpha]
.  Create it manually using the web interface following {url-az-project}[Microsoft official guide]

* When done manually, you will need to choose a workflow. Learn more about the different options in the {url-az-board}[official documentation].

. Create it in an automated way using the provided script, as shown below.

* When done with the script, you can add the flag `-p` to specify the workflow, if this flag is not set then the "Basic" workflow will be chosen. For more information about the different workflows see the following {url-az-workflows}[link].
 
== *Creating Azure DevOps project using provided script*

The script located at `scripts/accounts/azure-devops/create-project.sh` enables you to create a new Azure DevOps project or configure an existing one.

== Usage

```
create-project.sh \
  -n <name> \
  -d <description> \
  -o <organization> \
  -v <visibility> \
  -t <PAT> \
  [-p <workflow>]
```

== Flags

```
-n    [Required] Name of the new project.
-d    [Required] Description for the new project.

> -o    [Required] Name of the organization  for which the project will be configured.

Reviewer: ```suggestion
-o    [Required] Name of the organization for which the project will be configured.
```

> -v    [Required] Visibility. Accepted values: private or public.

Reviewer: ```suggestion
-v    [Required] Visibility. Accepted values: private, public.
```

-t    [Required] PAT token to login Azure DevOps.
-p               Process that will be used. Accepted values: basic, agile, scrum, cmmi.

```

== Example

> `./create-project.sh -n "Hello World" -d "This is a sample application" -o 'devon-hangar' -v 'public' -t 'myToken' -p 'agile'`
>    
> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion
`./create-project.sh -n "Hello World" -d "This is a sample application" -o devon-hangar -v public -t myToken -p agile`
```
